### PR TITLE
delete_question_scenarios

### PR DESCRIPTION
**Pull request by yash6102000**
if number of questions in test and  number of questions in assessment is same then throw proper error

### CODE REVIEW

File: app/components/common-components/DeletePopUp.tsx [hunk 1]
```diff
@@ -13,6 +13,7 @@ export default function DeletePopUp({
   status,
   deleteItemType,
```
Comment (anuragP04) [inline app/components/common-components/DeletePopUp.tsx:14]: why do we have change in this file?

File: app/components/sections/QuestionCard.tsx [hunk 2]
```diff
@@ -17,7 +16,10 @@ const QuestionCard = ({
   onAccordianToggle,
   index,
 }: {
-  question: Question & { questionType?: QuestionType }
+  question: Question & {
+    questionType?: QuestionType
+    // candidateQuestion?: CandidateQuestion[]
```
Comment (anuragP04) [inline app/components/sections/QuestionCard.tsx:21]: remove commented code

File: app/components/sections/QuestionCard.tsx [hunk 3]
```diff
@@ -17,7 +16,10 @@ const QuestionCard = ({
...
+  }
```
Comment (anuragP04) [inline app/components/sections/QuestionCard.tsx:22]: why do we have extra lines added here?

File: app/components/sections/SectionDetails.tsx [hunk 4]
```diff
@@ -12,6 +12,7 @@ const SectionDetails = () => {
   const { t } = useTranslation()
```
Comment (anuragP04) [inline app/components/sections/SectionDetails.tsx:12]: why do we have change in this file

File: app/routes/$workspaceId/tests/$sectionId.tsx [hunk 5]
```diff
@@ -59,13 +42,16 @@ export const action: ActionFunction = async ({ request, params }) => {
 export default function Section() {
   const { t } = useTranslation()
   const section = useActionData() as ActionData
+  console.log(section, 'section')
```
Comment (ayushicopods) [inline app/routes/$workspaceId/tests/$sectionId.tsx:45]: remove console

File: app/models/sections.server.ts [hunk 6]
```diff
@@ -211,7 +211,51 @@ export async function addQuestion(
     })
 }
 export async function deleteQuestionById(id: string) {
-  return prisma.question.update({
+  const getQuestion = await prisma.question.findUnique({
```
Comment (anuragP04) [inline app/models/sections.server.ts:214]: variable name is getQuestion???

File: app/models/sections.server.ts [hunk 7]
```diff
@@ -211,7 +211,51 @@ export async function addQuestion(
...
+        },
+      },
+    },
+  })
+  const totalQuestionArray = getQuestion?.section?.sectionInTest.map(
+    (data: any) => {
+      return data.totalQuestions
+    }
+  )
+  const sortTotalTime = totalQuestionArray?.sort((a: any, b: any) => {
```
Comment (anuragP04) [inline app/models/sections.server.ts:249]: avoid using any

Comment (anuragP04) [inline app/models/sections.server.ts:249]: why we are using time here to compare?

File: app/models/sections.server.ts [hunk 8]
```diff
@@ -211,7 +211,52 @@ export async function addQuestion(
...
+        },
+      },
+    },
+  })
+  const totalQuestionArray = questionData?.section?.sectionInTest.map(
+    (data: any) => {
+      return data.totalQuestions
+    }
+  )
+  console.log(totalQuestionArray, 'totalQuestionArray')
```
Comment (richa-copods) [inline app/models/sections.server.ts:249]: Remove console

File: app/models/sections.server.ts [hunk 9]
```diff
@@ -211,7 +211,52 @@ export async function addQuestion(
...
+            select: {
+              totalQuestions: true,
+            },
+          },
+        },
+      },
+    },
+  })
+  const totalQuestionArray = questionData?.section?.sectionInTest.map(
+    (data: any) => {
```
Comment (richa-copods) [inline app/models/sections.server.ts:245]: Give proper type. Avoid using any

File: app/routes/$workspaceId/tests/$sectionId.tsx [hunk 10]
```diff
@@ -34,23 +34,6 @@ export const action: ActionFunction = async ({ request, params }) => {
   const id = formData.get('id') as string
   if (action === 'deleteQuestion') {
```
Comment (richa-copods) [inline app/routes/$workspaceId/tests/$sectionId.tsx:35]: Use enum for action

File: app/routes/$workspaceId/tests/$sectionId.tsx [hunk 11]
```diff
@@ -60,11 +43,11 @@ export default function Section() {
   const { t } = useTranslation()
   const section = useActionData() as ActionData
   useEffect(() => {
-    if (section?.resp?.status === 200) {
-      toast.success(t(section.resp?.title))
-    } else if (section?.errors?.status === 400) {
-      toast.error(t(section.errors?.title), {
-        toastId: section.errors?.title,
+    if ((section && (section as string)) === 'success') {
```
Comment (richa-copods) [inline app/routes/$workspaceId/tests/$sectionId.tsx:46]: Write condition properly. This is not standard.

File: app/routes/$workspaceId/tests/$sectionId.tsx [hunk 12]
```diff
@@ -60,11 +43,11 @@ export default function Section() {
...
+      toast.success(t('statusCheck.deletedSuccess'))
+    } else if ((section && (section as string)) === 'not deleted') {
```
Comment (richa-copods) [inline app/routes/$workspaceId/tests/$sectionId.tsx:48]: Write condition properly. This is not standard.

File: public/locales/en/index.json [hunk 13]
```diff
@@ -143,7 +143,9 @@
     "msq": "MSQ",
     "mcq": "MCQ",
     "multipleSelectQuestion": "Multiple Select Question",
-    "multipleChoiceQuestion": "Multiple Choice Question"
+    "multipleChoiceQuestion": "Multiple Choice Question",
+    "testMayAffectWarning": "This question is being used in a test, further invites will not get this question.",
```
Comment (richa-copods) [inline public/locales/en/index.json:147]: Use existing message of "Deleted Successfully..!". No need of this

File: public/locales/en/index.json [hunk 14]
```diff
@@ -143,7 +143,9 @@
...
+    "questionNotDeleted": "Cannot delete question. This test is used in assessment and will cause conflict after deleting."
```
Comment (richa-copods) [inline public/locales/en/index.json:148]: Test can not be deleted as being used in existing Assessment.

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/components/sections/QuestionCard.tsx (hunk 3 above)
Comment (richa-copods) [reply app/components/sections/QuestionCard.tsx:17]: @yash6102000 acknowledge this please

File: app/components/sections/SectionDetails.tsx (hunk 4 above)
Comment (richa-copods) [reply app/components/sections/SectionDetails.tsx:12]: @yash6102000 acknowledge this please

File: app/models/sections.server.ts [hunk 15]
```diff
@@ -220,4 +265,5 @@ export async function deleteQuestionById(id: string) {
       deletedAt: new Date().toString(),
     },
   })
+  return deleteQuestion && 'success'
```
Comment (richa-copods) [inline app/models/sections.server.ts:268]: Return simple `true` or `false`

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/routes/$workspaceId/tests/$sectionId.tsx [hunk 16]
```diff
@@ -32,39 +23,22 @@ export const action: ActionFunction = async ({ request, params }) => {
...
   const { t } = useTranslation()
-  const section = useActionData() as ActionData
+  const section = useActionData()
   useEffect(() => {
-    if (section?.resp?.status === 200) {
-      toast.success(t(section.resp?.title))
-    } else if (section?.errors?.status === 400) {
-      toast.error(t(section.errors?.title), {
-        toastId: section.errors?.title,
+    if (section === true) {
```
Comment (richa-copods) [inline app/routes/$workspaceId/tests/$sectionId.tsx:37]: use condition `if(section)`

File: app/routes/$workspaceId/tests/$sectionId.tsx [hunk 17]
```diff
@@ -32,39 +23,22 @@ export const action: ActionFunction = async ({ request, params }) => {
...
+      toast.success(t('statusCheck.deletedSuccess'))
+    } else if (section === false) {
```
Comment (richa-copods) [inline app/routes/$workspaceId/tests/$sectionId.tsx:39]: use `else` instead of `else if`

File: public/locales/en/index.json [hunk 18]
```diff
@@ -143,7 +143,8 @@
     "msq": "MSQ",
     "mcq": "MCQ",
     "multipleSelectQuestion": "Multiple Select Question",
-    "multipleChoiceQuestion": "Multiple Choice Question"
+    "multipleChoiceQuestion": "Multiple Choice Question",
+    "questionNotDeleted": "Test can not be deleted as being used in existing Assessment."
```
Comment (richa-copods) [inline public/locales/en/index.json:147]: change to "Question cannot be deleted..."

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/components/sections/QuestionCard.tsx [hunk 19]
```diff
@@ -1,27 +1,20 @@
-import { useState } from "react"
-
 import { Icon } from "@iconify/react"
-import { useSubmit } from "@remix-run/react"
 import { useTranslation } from "react-i18next"
```
Comment (richa-copods) [inline app/components/sections/QuestionCard.tsx]: Resolve all the ESLint warnings

File: app/routes/$workspaceId/tests/$sectionId.tsx (hunk 17 above)
Comment (anuragP04) [reply app/routes/$workspaceId/tests/$sectionId.tsx:40]: @yash6102000 , please resolve this comment, it is not resolved yet. If you have any doubts or need clarity please let us know.

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/models/sections.server.ts [hunk 20]
```diff
@@ -212,7 +212,50 @@ export async function addQuestion(
...
+  const totalQuestionArray = questionData?.section?.sectionInTest.map(
+    (data: { totalQuestions: number }) => {
+      return data.totalQuestions
+    }
+  )
+  if (questionData && totalQuestionArray) {
+    if (
+      questionData.section.questions?.length <= Math.max(...totalQuestionArray)
+    ) {
+      return "not deleted"
```
Comment (richa-copods) [inline app/models/sections.server.ts:254]: Hey change it to "notDeleted". And better if you create ENUM for this

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/interface/Interface.tsx [hunk 21]
```diff
@@ -305,6 +306,10 @@ export enum QuestionTypes {
   singleChoice = "SINGLE_CHOICE",
   text = "TEXT",
 }
+export enum deleteQuestionStatus {
+  questionNotDeleted = "QUESTION_NOT_DELETED",
```
Comment (richa-copods) [inline app/interface/Interface.tsx:310]: You can simply keep it as `notDeleted = NOT_DELETED`, as ENUM specifies it is status for delete question

File: app/routes/$workspaceId/tests/$sectionId.tsx [hunk 22]
```diff
@@ -34,39 +26,23 @@ export const action: ActionFunction = async ({ request, params }) => {
...
-    if (section?.resp?.status === 200) {
-      toast.success(t(section.resp?.title))
-    } else if (section?.errors?.status === 400) {
-      toast.error(t(section.errors?.title), {
-        toastId: section.errors?.title,
+    if (section === deleteQuestionStatus.questionDeleted) {
+      toast.success(t("statusCheck.deletedSuccess"))
+    } else if (section === deleteQuestionStatus.questionNotDeleted) {
+      toast.error(t("sectionsConstants.questionNotDeleted"), {
+        toastId: "sectionsConstants.questionNotDeleted",
```
Comment (richa-copods) [inline app/routes/$workspaceId/tests/$sectionId.tsx:45]: Give proper toastId.

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/models/sections.server.ts [hunk 23]
```diff
@@ -212,7 +213,50 @@ export async function addQuestion(
...
+            },
+            select: {
+              totalQuestions: true,
+            },
+          },
+        },
+      },
+    },
+  })
+  const totalQuestionArray = questionData?.section?.sectionInTest.map(
```
Comment (richa-copods) [inline app/models/sections.server.ts:246]: Rename it to totalQuestionsList

**Review by richa-copods — CHANGES_REQUESTED** [review]

**Pull request merged**